api/tests/security/csrf/test_csrf.py

```diff
@@ -117,6 +98,12 @@ def test_parse_csrf_token(mock_urandom, mock_csrf_token_value):
 
     assert actual_token_value == mock_csrf_token_value
 
+def test_parse_csrf_token_expired(mock_urandom):
+    csrf_token = generate_csrf_token(SECRET_KEY,SALT)
+    time.sleep(1)
+
+    with pytest.raises(BadSignature):
+        parse_csrf_token(SECRET_KEY, SALT, csrf_token, max_age=0)
```

Comment:
question: how does this work? validation fails if the signature is older than max_age=0? can we use a mechanism other than using an actual sleep? (like fake timers)

Reply:
I already looked for fake timers but thery're not included in the pytest suit, so we would need another dependency and I thought that for this single instance it was not worth it

Reply:
I would inject a timeprovider or something similar that can be mocked, and use that to request the time on the parse function.

Reply:
Unfortunately there's not one out of the box. We would need another library like [freezegun](https://github.com/spulec/freezegun)